src/ResourceManager/Sql/Commands.Sql/Elastic Jobs/Services/AzureSqlElasticJobAdapter.cs

```diff
@@ -0,0 +1,1164 @@
+        /// <param name="serverName">The server name</param>
+        /// <param name="agentName">The agent name</param>
+        /// <param name="credentialName">The job credential name</param>
+        /// <returns>The job credential object</returns>
+        public AzureSqlElasticJobCredentialModel GetJobCredential(string resourceGroupName, string serverName, string agentName, string credentialName)
+        {
+            var resp = Communicator.GetJobCredential(resourceGroupName, serverName, agentName, credentialName);
+            return CreateAgentCredentialModelFromResponse(resourceGroupName, serverName, agentName, resp);
+        }
+
+        /// <summary>
+        /// Gets a list of job credentials from the control database
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server name</param>
+        /// <param name="agentName">The agent name</param>
+        /// <returns>A list of job credentials</returns>
+        public List<AzureSqlElasticJobCredentialModel> ListJobCredentials(string resourceGroupName, string serverName, string agentName)
+        {
+            var resp = Communicator.GetJobCredential(resourceGroupName, serverName, agentName);
+            return resp.Select(credentialName => CreateAgentCredentialModelFromResponse(resourceGroupName, serverName, agentName, credentialName)).ToList();
+        }
+
+        /// <summary>
+        /// Deletes a job credential from the control database
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server the agent is in</param>
+        /// <param name="agentName">The agent name</param>
+        /// <param name="credentialName">The job credential name</param>
+        public void RemoveJobCredential(string resourceGroupName, string serverName, string agentName, string credentialName)
+        {
+            Communicator.RemoveJobCredential(resourceGroupName, serverName, agentName, credentialName);
+        }
+
+        /// <summary>
+        /// Converts a job credential response to a AzureSqlElasticJobCredentialModel object
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server name</param>
+        /// <param name="agentName">The agent name</param>
+        /// <param name="resp">The job credential response</param>
+        /// <returns></returns>
+        private static AzureSqlElasticJobCredentialModel CreateAgentCredentialModelFromResponse(string resourceGroupName, string serverName, string agentName, JobCredential resp)
+        {
+            // Parse credential name from id
+            // This is not expected to ever fail, but in case we have a bug here it's better to provide a more detailed error message
+            int lastSlashIndex = resp.Id.LastIndexOf('/');
```

Comment:
Can you use `new ResourceIdentifier(...).ResourceName` like you did below?

Reply:
Will update.